pkg/datapath/linux/node.go

```diff
@@ -1196,34 +1296,60 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *nodeTypes.Node, firstAdd
 		// assumptions:
 		// 1. newNode is accessed only by reads.
 		// 2. It is safe to invoke insertNeighbor for the same node.
+		//
+		// Because neighbor inserts is not synced, we do not currently
+		// collect/ bubble up errors.
+		// Instead we just rely on logging errors as they come up in the
+		// neighbor update procedure.
+		//
+		// In v1.15, we will have the neighbor sync component report its own
+		// health via stored errors.
 		go n.insertNeighbor(context.Background(), newNode, false)
 	}
 
 	if n.nodeConfig.EnableIPSec && n.nodeConfig.EncryptNode && !n.subnetEncryption() {
-		n.encryptNode(newNode)
+		if err := n.encryptNode(newNode); err != nil {
+			acc = errors.Join(acc, fmt.Errorf("encrypt node: %w", err))
+		}
 	}
 
+	// Local node update
 	if newNode.IsLocal() {
 		isLocalNode = true
 		if n.nodeConfig.EnableLocalNodeRoute {
-			n.updateOrRemoveNodeRoutes(oldAllIP4AllocCidrs, newAllIP4AllocCidrs, isLocalNode)
-			n.updateOrRemoveNodeRoutes(oldAllIP6AllocCidrs, newAllIP6AllocCidrs, isLocalNode)
+			if err := n.updateOrRemoveNodeRoutes(oldAllIP4AllocCidrs, newAllIP4AllocCidrs, isLocalNode); err != nil {
+				acc = errors.Join(acc, fmt.Errorf("enable local node route: update ipv4 routes: %w", err))
+			}
+			if err := n.updateOrRemoveNodeRoutes(oldAllIP6AllocCidrs, newAllIP6AllocCidrs, isLocalNode); err != nil {
+				acc = errors.Join(acc, fmt.Errorf("enable local node route: update ipv6 routes: %w", err))
+			}
 		}
 		if n.subnetEncryption() {
-			n.enableSubnetIPsec(n.nodeConfig.IPv4PodSubnets, n.nodeConfig.IPv6PodSubnets)
+			// Enables subnet IPSec by upserting node host routing table IPSec routing
+			if err := n.enableSubnetIPsec(n.nodeConfig.IPv4PodSubnets, n.nodeConfig.IPv6PodSubnets); err != nil {
+				acc = errors.Join(acc, fmt.Errorf("subnet encryption: %w", err))
+			}
 		}
 		if firstAddition && n.nodeConfig.EnableIPSec {
 			n.registerIpsecMetricOnce()
 		}
-		return nil
+		return acc
 	}
 
 	if n.nodeConfig.EnableAutoDirectRouting {
-		n.updateDirectRoutes(oldAllIP4AllocCidrs, newAllIP4AllocCidrs, oldIP4, newIP4, firstAddition, n.nodeConfig.EnableIPv4)
-		n.updateDirectRoutes(oldAllIP6AllocCidrs, newAllIP6AllocCidrs, oldIP6, newIP6, firstAddition, n.nodeConfig.EnableIPv6)
-		return nil
+		if err := func() error {
```

Comment:
The reasoning here, as well as the separate errors, is to prevent duplication when a series of function calls are all to be similarly "scoped".

For example

```go
errs = errors.Join(errs, fmt.Errorf("tried some thing: %w", foo1()))
//...
errs = errors.Join(errs, fmt.Errorf("tried some thing: %w", foo2()))
//...
errs = errors.Join(errs, fmt.Errorf("tried some thing: %w", foo3()))
// ...
```

Perhaps the closure here is a bit out of place, I didn't want to go and start refactoring sections of this into separate functions so I was trying to settle between the three approaches you see (somewhat inconstantly):

* create separate error accumulator.
* use a closure, wrap one error
* duplicate the same string around.

I'll try to make this more consistent for now,  I think we can go with "good enough" with approach three for now.

Reply:
Cleaned this up to be more consistent with the rest of the changes.